workflows/plots.py

```diff
@@ -237,69 +237,103 @@ def plot_sfs(s, outfile):
-    #plt.ylim([0,0.04])
-    #plt.title('dadi')
-
-    plt.subplot(2,2,3)
-    plt.ylabel('Shape parameter')
-    plt.xlabel('Population')
-    plt.boxplot([polydfe_pop0['b'], polydfe_pop1['b'], polydfe_pop2['b']])
-    plt.plot(plt.xlim(), [0.19,0.19], c='red')
-    plt.xticks([1,2,3],['YRI','CEU','CHB'])
-    plt.ylim([0.1,0.25])
-    plt.title('polyDFE')
-
-    plt.subplot(2,2,4)
-    plt.ylabel('|E(s)|')
-    plt.xlabel('Population')
-    plt.boxplot([polydfe_pop0_Es, polydfe_pop1_Es, polydfe_pop2_Es])
-    plt.plot(plt.xlim(), [0.014,0.014], c='red')
-    plt.xticks([1,2,3],['YRI','CEU','CHB'])
-    #plt.ylim([0,0.04])
-    plt.title('polyDFE')
+    if xtick_label is None: xtick_label = ['pop'+str(i) for i in np.arange(len(dadi_shapes))]
+    plt.subplot(3,2,1)
+    _plot_boxplots(plt, 'dadi', dadi_shapes, 0.19, xtick_label, 'Shape', [0.1,0.25])
+    plt.subplot(3,2,2)
+    _plot_boxplots(plt, 'dadi', dadi_Es, 0.014, xtick_label, '|E(s)|')
+
+    plt.subplot(3,2,3)
+    _plot_boxplots(plt, 'polyDFE', polydfe_shapes, 0.19, xtick_label, 'Shape', [0.1,0.25])
+    plt.subplot(3,2,4)
+    _plot_boxplots(plt, 'polyDFE', polydfe_Es, 0.014, xtick_label, '|E(s)|')
+
+    plt.subplot(3,2,5)
+    _plot_boxplots(plt, 'DFE-alpha', dfe_alpha_shapes, 0.19, xtick_label, 'Shape')
+    plt.subplot(3,2,6)
+    _plot_boxplots(plt, 'DFE-alpha', dfe_alpha_Es, 0.014, xtick_label, '|E(s)|')
 
     fig.tight_layout()
     plt.savefig(output[0])
+
+def _read_bestfits(input_files):
+    """
+    Description:
+        Combines all input files into a single data frame,
+        then separates the single data frame into several data frames based on population ids.
+
+    Arguments:
+        input_files list: A list containing input files. 
+
+    Returns:
+        bestfits list: A list containing data frames with bestfit parameters for different populations. 
+    """
+    pop_ids = ['pop'+str(i) for i in np.arange(len(input_files))]
+    df_from_each_file = (pd.read_csv(f, sep="\t") for f in input_files)
+    concatenated_df   = pd.concat(df_from_each_file, ignore_index=True)
+    bestfits = [concatenated_df[concatenated_df['pop_id'] == p] for p in pop_ids]
+
+    return bestfits
+
+def _plot_boxplots(plt, title, data, tval, xtick_label, ylabel, ylim=None):
+    """
+    Description:
+        Plots box plots for bestfit results from three populations.
+
+    Arguments:
+        plt matplotlib.pyplot: Interface to matplotlib. 
+        title str: Title of the boxplot.
+        data list: Bestfit results to be plotted. 
+        tval float: True value for the parameter.
+        xtick_label list: X-axis tick labels.
+        ylabel str: Y-axis label.
+        ylim list: Y-axis limits.
+    """
+    plt.ylabel(ylabel)
+    plt.xlabel('Population')
+    plt.boxplot(data)
+    plt.plot(plt.xlim(), [tval,tval], c='red')
+    plt.xticks(np.arange(len(data))+1,xtick_label)
+    if ylim is not None: plt.ylim(ylim)
+    plt.title(title)
```

Comment:
@andrewkern here is the function for plotting box plots, now it can work with a single population

Reply:
cool. check my note above on pulling the population names directly from the demographic models